### PR TITLE
Allow custom build parallelism

### PR DESCRIPTION
**Pull request by t0yv0**
Same change as just got rolled out to ecosystem providers https://github.com/pulumi/ci-mgmt/pull/386 - applied manually to this provider.

### CODE REVIEW

**Review by iwahbe — APPROVED** [review]
Looks good from my end. We should probably make sure that someone from @pulumi/providers gives it the 👌.

File: Makefile [hunk 1]
```diff
@@ -13,6 +13,7 @@ JAVA_GEN_VERSION := v0.7.1
 TESTPARALLELISM := 10
 WORKING_DIR := $(shell pwd)
 SCHEMA_TOOLS_VERSION := $(shell cat .schema-tools.version)
+PULUMI_PROVIDER_BUILD_PARALLELISM ?= -p 2
```
Comment (t0yv0) [inline Makefile:16]: This is a behavior change from -p 1 but @thomas11 recommended this. We can make it -p 1. Still lets me run with -p 6 locally.

**Comment by t0yv0** [comment]
b/sdk/go/aws/cognito/identityPoolRoleAttachment.go strikes again, interesting.

File: Makefile (hunk 1 above)
Comment (rquitales) [inline Makefile:16]: Should we maintain consistency with `TESTPARALLELISM`, where we define the value only, eg `2` instead of `-p 2`?

**Review by rquitales — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (rquitales) [reply Makefile:16]: NVM. Just refferred to https://github.com/pulumi/ci-mgmt/pull/386 and including the flag within the variable makes it easier to unset the flag if required.

**Review by rquitales — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (t0yv0) [reply Makefile:16]: This is something I'd like in my .bashrc hence the longer name PULUMI_PROVIDER_BUILD_PARALLELISM

**Comment by t0yv0** [comment]
b/sdk/go/aws/cognito/identityPoolRoleAttachment.go  is flaky. rerunning fixed it.

**Pull request merged**